lib/make-issue-token-route.ts

```diff
@@ -0,0 +1,66 @@
+import { ResponseToolkit, Request } from "@hapi/hapi";
+import querystring from "querystring";
+import Wreck from "@hapi/wreck";
+import btoa from "btoa";
+import { ClientSecrets, TokenPayload } from ".";
+
+type RequestWithPossibleErrorResponse = Request & {
+  response: {
+    isBoom?: boolean;
+    data?: object;
+    output?: {
+      payload?: { [key: string]: any };
+    };
+  };
+};
+
+const token = (clients: ClientSecrets, tokenEndpoint: string) => async (
+  payload: TokenPayload
+) => {
+  const { client_id, ...tokenPayload } = payload;
+
+  const options = {
+    headers: {
+      Authorization: "Basic " + btoa(`${client_id}:${clients[client_id]}`),
+      "Content-Type": "application/x-www-form-urlencoded",
+    },
+    //@ts-ignore weird types for querystring
+    payload: querystring.stringify(tokenPayload),
```

Comment:
idk, i can't figure querystring type out and this feels like the exact kind of thing I don't want to be worrying about in ts

Reply:
It is complaining because `querystring` isn't able to infer that `tokenPayload` doesn't include properties that are objects (`ParsedUrlQueryInput` extends `NodeJs.Dict<"querystringable stuff">`).

Casting it suppresses the error, even though it seems like it should be unnecessary.

```suggestion
    payload: querystring.stringify(tokenPayload as querystring.ParsedUrlQueryInput),
```

Reply:
I guess there are [reasons] https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38414. )

But to your point, you don't need to worry about them. When I know something is fine, I'd cast it as opposed to ignore.

Reply:
Gotcha